libs/hdf-converters/src/checklist-mapper.ts

```diff
@@ -0,0 +1,454 @@
+ * @param data
+ * @returns checklist
+ * @experimental
+ */
+function revertChecklist(data: Record<string, unknown>): string {
+  const context = new Jsonix.Context([checklistMapping.jsonixMapping]);
+
+  const marshaller = context.createMarshaller();
+  return marshaller.marshalString(data);
+}
+
+// Status mapping for going to and from checklist
+const STATUS_MAPPING: Map<string | undefined, string> = new Map([
+  ['NotAFinding', 'Passed'],
+  ['Open', 'Failed'],
+  ['Not_Applicable', 'Not Applicable'],
+  ['Not_Reviewed', 'Not Reviewed'],
+  ['Passed', 'NotAFinding'],
+  ['Failed', 'Open'],
+  ['Not Applicable', 'Not_Applicable'],
+  ['Not Reviewed', 'Not_Reviewed']
+]);
+
+/**
+ * Checklist converter - not implemented and may be used in
+ * conjunction with revertChecklist (also experimental)
+ * @experimental
+ */
+export class ChecklistConverter {
+  static toChecklist(data: ChecklistFile): string {
+    // Updating assets
+    const asset = {...data.asset, TYPE_NAME: 'Checklist.ASSET'};
+    _.set(data, 'raw.value.asset', asset);
+
+    // Updating marked-up rule data
+    const iStigs: ChecklistStig[] = _.get(
+      data.raw,
+      'value.stigs.istig'
+    ) as ChecklistStig[];
+    iStigs.forEach((stig: any, stig_index: number) => {
+      _.get(stig, 'vuln').forEach((vuln: any, vuln_index: number) => {
+        _.set(
+          vuln,
+          'status',
+          STATUS_MAPPING.get(data.stigs[stig_index].vulns[vuln_index].status)
+        );
+        _.set(
+          vuln,
+          'findingdetails',
+          data.stigs[stig_index].vulns[vuln_index].findingDetails
+        );
+        _.set(
+          vuln,
+          'comments',
+          data.stigs[stig_index].vulns[vuln_index].comments
+        );
+        _.set(
+          vuln,
+          'severityoverride',
+          data.stigs[stig_index].vulns[vuln_index].severityOverride
+        );
+        _.set(
+          vuln,
+          'severityjustification',
+          data.stigs[stig_index].vulns[vuln_index].severityJustification
+        );
+      });
+    });
+
+    return revertChecklist(data.raw);
+  }
+}
+
+const IMPACT_MAPPING: Map<string, number> = new Map([
+  ['high', 0.7],
+  ['medium', 0.5],
+  ['low', 0.3],
+  ['none', 0.0]
+]);
+
+/**
+ * Tranformer function that splits a string and return array
+ * @param input - string of CCI references
+ * @returns ref - array of CCI references
+ */
+function cciRef(input: string): string[] {
+  return input.split('; ');
+}
+
+const CCI_NIST_MAPPING = new CciNistMapping();
+
+/**
+ * Transformer function that splits string and maps resulting array
+ * into NIST control tags
+ * @param input - string of CCI references
+ * @returns tag - array of NIST Control Tags
+ */
+function nistTag(input: string): string[] {
+  const identifiers: string[] = input.split('; ');
+  return CCI_NIST_MAPPING.nistFilter(
+    identifiers,
+    DEFAULT_STATIC_CODE_ANALYSIS_NIST_TAGS
```

Comment:
ask @ejaronne if the default static code analysis nist tags are what we ought to use for checklists